tests/block_header_test.go

```diff
@@ -107,8 +110,23 @@ func TestBlockHeader_SatisfiesInvariants(t *testing.T) {
 		t.Run("LastBlockOfEpochContainsSealingTransaction", func(t *testing.T) {
 			testHeaders_LastBlockOfEpochContainsSealingTransaction(t, headers, client)
 		})
+	}
+
+	runTests()
+	require.NoError(net.Restart())
+	runTests()
+
+	// TODO: enable when genesis is fixed
+	// require.NoError(net.restartWithExportImport())
+	// runTests()
+}
+
+func testHeaders_CompareHeadersHashes(t *testing.T, hashes []common.Hash, newHeaders []*types.Header) {
+	require := require.New(t)
 
-		require.NoError(net.Restart())
+	require.Len(newHeaders, len(hashes), "length mismatch")
```

Comment:
This test may be flaky since new (empty) blocks may have been created since the restart.

It should be enough to check that `len(newHeaders) >= len(hashes)` and then just test the first `len(hashes)` headers.

Reply:
yes, you are right